src/NetscapeBookmarkParser.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shaarli\NetscapeBookmarkParser;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Shaarli\NetscapeBookmarkParser\DataFormatter\BookmarksDataFormatter;
+use Shaarli\NetscapeBookmarkParser\Encoder\NetscapeBookmarkDecoder;
+use Shaarli\NetscapeBookmarkParser\Encoder\NetscapeBookmarkEncoder;
+
+/**
+ * Generic Netscape bookmark parser
+ */
+class NetscapeBookmarkParser
+{
+    public const DEPRECATION_VERSION = '4.0';
+
+    private $defaultContext = [
+        NetscapeBookmarkDecoder::KEEP_NESTED_TAGS => true,
+        NetscapeBookmarkDecoder::NORMALIZE_DATES  => true,
+        NetscapeBookmarkDecoder::DATE_RANGE       => '30 years',
+    ];
+
+    /**
+     * @var LoggerInterface instance.
+     */
+    private $logger;
+
+    /**
+     * @var NetscapeBookmarkDecoder
+     */
+    private $decoder;
+
+    /**
+     * @var BookmarksDataFormatter
+     */
+    private $formatter;
+
+    /**
+     * @var NetscapeBookmarkEncoder
+     */
+    private $encoder;
+
+    public function __construct(array $defaultContext = [], LoggerInterface $logger = null)
+    {
+        $this->defaultContext = array_merge($this->defaultContext, $defaultContext);
+        $this->logger = $logger ?? new NullLogger();
+
+        $this->decoder = new NetscapeBookmarkDecoder($this->defaultContext, $this->logger);
+        $this->formatter = new BookmarksDataFormatter();
+        $this->encoder = new NetscapeBookmarkEncoder();
+    }
+
+    /**
+     * Parses a Netscape bookmark file
+     *
+     * @param string $filePath Bookmark file to parse
+     *
+     * @return array An associative array containing parsed links
+     */
+    public function parseFile(string $filePath): array
+    {
+        $this->logger->info('Starting to parse ' . $filePath);
+
+        try {
+            return $this->decoder->decode(file_get_contents($filePath));
+        } catch (\Exception $exception) {
+            $this->logger->error($exception);
```

Comment:
It should throw an exception, eventually a custom one. Let's stop silencing errors :D

Reply:
I removed all `try catch` blocks